sympy/solvers/solvers.py

```diff
@@ -2051,8 +2071,12 @@ def key(sym):
                     if b in result:
                         result.remove(b)
 
-    default_simplify = bool(failed)  # rely on system-solvers to simplify
-    if  flags.get('simplify', default_simplify):
+    if not result:
+        return []
+
+    # rely on linear system solvers to simplify
+    default_simplify = bool(failed) or not linear
+    if flags.get('simplify', default_simplify):
```

Comment:
This seems to be the problem. Why has `default_simplify` been changed?

Reply:
This was changed because the nonlinear solutions were not being simplified.